vmware-tanzu/cartographer
# Cartotest supplychains

[PR requirements]: https://github.com/vmware-tanzu/cartographer/blob/main/CONTRIBUTING.md#commit-message-and-pr-guidelines

## Changes proposed by this PR


Large rewrite of Cartotest
- Allow cartotest to accept supplychains alongside templates and workloads.
- Use abstraction to clarify the interface
- Add comparison option functions (for example, coercing numbers in expected and actual to be the same type)




## Release Note


[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative
Cartotest Improvement: Users may specify an actual supply chain or set of supply chain set when testing that the proper object is stamped.

!Breaking changes to CartoTest! CartoTest accepts interfaces rather than multiple concrete instances providing the same functionality. Field names have been shortened (e.g. TemplateTestCase --> Test). See changes to tests/templates/template_test.go for examples of rewriting existing tests to the new API.

## Cherry-pick branches
All minor branches. Has not been tested on previous minor branches.



## PR Checklist

Note: Please do not remove items. Mark items as done `[x]` or use ~strikethrough~ if you believe they are not relevant

- [ ] Linked to a relevant issue. Eg: `Fixes #123` or `Updates #123`
- [ ] Removed non-atomic or `wip` commits
- [x] Filled in the [Release Note](#Release-Note) section above
- [x] Added any relevant branches to cherry-pick
- [x] Modified the docs to match changes
